src/seaport.ts

```diff
@@ -1029,24 +1031,38 @@ export class Seaport {
     ]);
 
     const ordersMetadata: FulfillOrdersMetadata = fulfillOrderDetails.map(
-      (orderDetails, index) => ({
-        order: orderDetails.order,
-        unitsToFill: orderDetails.unitsToFill,
-        orderStatus: scaleOrderStatusToMaxUnits(
-          orderDetails.order,
-          orderStatuses[index],
-        ),
-        offerCriteria: orderDetails.offerCriteria ?? [],
-        considerationCriteria: orderDetails.considerationCriteria ?? [],
-        tips:
-          orderDetails.tips?.map((tip) => ({
-            ...mapInputItemToOfferItem(tip),
-            recipient: tip.recipient,
-          })) ?? [],
-        extraData: orderDetails.extraData ?? "0x",
-        offererBalancesAndApprovals: offerersBalancesAndApprovals[index],
-        offererOperator: allOffererOperators[index],
-      }),
+      (orderDetails, index) => {
+        const o = {
+          order: orderDetails.order,
+          unitsToFill: orderDetails.unitsToFill,
+          orderStatus: scaleOrderStatusToMaxUnits(
+            orderDetails.order,
+            orderStatuses[index],
+          ),
+          offerCriteria: orderDetails.offerCriteria ?? [],
+          considerationCriteria: orderDetails.considerationCriteria ?? [],
+          tips:
+            orderDetails.tips?.map((tip) => ({
+              ...mapInputItemToOfferItem(tip),
+              recipient: tip.recipient,
+            })) ?? [],
+          extraData: orderDetails.extraData ?? "0x",
+          offererBalancesAndApprovals: offerersBalancesAndApprovals[index],
+          offererOperator: allOffererOperators[index],
+        };
+        if (o.tips.length > 0) {
+          o.tips = adjustTipsForPartialFills(
+            o.tips,
+            o.unitsToFill || 1,
```

Comment:
I don't 100% understand this code (the type is bigint | undefined), but I'm making the assumption here if unitsToFill is undefined it's because it's a ERC721 sell order where units to fill is 1. Tests seem to agree.